cmd/podman/play_kube.go

```diff
@@ -117,12 +118,18 @@ func playKubeYAMLCmd(c *cliconfig.KubePlayValues) error {
 	podOptions = append(podOptions, libpod.WithInfraContainerPorts(podPorts))
 
 	// Create the Pod
-	pod, err := runtime.NewPod(ctx, podOptions...)
+	pod, err = runtime.NewPod(ctx, podOptions...)
 	if err != nil {
 		return err
 	}
-	// Print the Pod's ID
-	fmt.Println(pod.ID())
+	// We've now created a resource that will need to be cleaned up if we fail.
+	defer func() {
+		if err != nil {
```

Comment:
The `err` in the loop starting in line 193 will be ignored as it's in a new scope (see minimal example https://play.golang.com/p/JMJT9T0Fbj6). Although I find the solution with deferred checks elegant, I think they are prone to errors.

I prefer to add a helper function and wrap around that with an error check. This way, we don't have to worry about future changes that potentially break the deferred check.

```golang
if err := helperFunc(...); err != nil {
   logrus.Debugf(err.Error())
   if errCleanUp := runtime.RemovePod(...); errCleanup != nil {
     logrus.Errorf(....)
  }
}